src/lib/Provider/GoogleMapProvider.js

```diff
@@ -146,6 +152,99 @@ export default class GoogleMapProvider {
 		this.stopMarkers = [];
 	}
 
+	addVehicleMarker(vehicle, activeTrip) {
+		if (!this.map) return null;
+
+		// TODO: Add enum for vehicle color and status
+		let color;
+		if (!vehicle.predicted) {
+			color = COLORS.VEHICLE_REAL_TIME_OFF;
+		}
+
+		const busIcon = createVehicleIconSvg(vehicle?.orientation, color);
+		const icon = {
+			url: `data:image/svg+xml;charset=UTF-8,${encodeURIComponent(busIcon)}`,
+			scaledSize: new google.maps.Size(40, 40),
+			anchor: new google.maps.Point(20, 20)
+		};
+
+		const marker = new google.maps.Marker({
+			position: { lat: vehicle.position.lat, lng: vehicle.position.lon },
+			map: this.map,
+			icon: icon,
+			zIndex: 1000
+		});
+
+		this.vehicleMarkers.push(marker);
+
+		const vehicleData = {
+			nextDestination: activeTrip.tripHeadsign,
+			vehicleId: vehicle.vehicleId,
+			lastUpdateTime: vehicle.lastUpdateTime,
+			nextStopName: this.stopsMap.get(vehicle.nextStop)?.name || 'N/A',
```

Comment:
is `'N/A'` the most appropriate value for this? It's ok if it is, but it should be i18n'd.